### ISSUE TITLE
Comparing lists that contain themselves causes stack overflow

### ISSUE THREAD

**Issue by rdococ** [jmoenig/Snap#2378]
![comparing self-referential lists](https://user-images.githubusercontent.com/3169568/56391953-085c6e80-6228-11e9-92ee-b3107d7ce751.png)

The error I get is "maximum call stack size exceeded". I'm guessing that the equals operator is comparing each item of the two (in this case, one) lists recursively, and because both lists are infinitely nested, it just keeps comparing over and over.

**Comment by brianharvey** [comment]
Yeah.  We really don't support circular lists.  Equals is a primitive, but user code is likely to overflow too.  Maybe someday when things slow down.

**Comment by rdococ** [comment]
Until then, maybe it could give a different error message, such as "Error: One or both of the lists being compared is recursive"?
